cypress/integration/CreateModels/EndSession.spec.js

```diff
@@ -0,0 +1,102 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+import * as models from '../../support/Models'
+import * as modelPage from '../../support/components/ModelPage'
+import * as actions from '../../support/Actions'
+import * as editDialogModal from '../../support/components/EditDialogModal'
+import * as train from '../../support/Train'
+import * as trainDialogsGrid from '../../support/components/TrainDialogsGrid'
+import * as helpers from '../../support/Helpers'
+
+const preliminaryTrainingDescription = 'Preliminary Training to cause some expected behaviors in future Train Dialogs'
+
+describe('End Session - Create Model', () => {
+  afterEach(helpers.SkipRemainingTestsOfSuiteIfFailed)
+  
+  context('Should create a new model', () => {
+    it('Create a model to test against', () => {
+      models.CreateNewModel('z-EndSession')
+    })
+
+    it('Should create three Actions', () => {
+      actions.CreateNewActionThenVerifyInGrid({ response: "Hello" })
+      actions.CreateNewActionThenVerifyInGrid({ response: "Okay" })
+      actions.CreateNewActionThenVerifyInGrid({ type: 'END_SESSION', response: "Goodbye" })
+    })
+  })
+
+  context('1st Train Dialog', () => {
+    it('Should create a new Train Dialog', () => {
+      modelPage.NavigateToTrainDialogs()
+      cy.WaitForTrainingStatusCompleted()
+      train.CreateNewTrainDialog()
+    })
+
+    it('Should train model to respond to "Hi"', () => {
+      train.TypeYourMessage('Hi')
+      editDialogModal.ClickScoreActionsButton()
+      train.SelectAction('Hello')
+    })
+
+    it('Should train model to respond to "Yo"', () => {
+      train.TypeYourMessage('Yo')
```

Comment:
Again on chose this as example but the pattern is in all the tests. It uses different helpers `train` then `editDialogModal` then `train`. The writer would expect to use the same "helper" for the same modal or part of the app.  This doesn't seem very intuitive.  I think people would only do this since the methods they want to use don't exist on the other object and they have to. 

Eventually I would like to see these reorganized so there isn't a switch or a better explanation as to why their needs to be a switch. I would expect if `train` is for operating on single train dialog it could also be used for editing.

Reply:
Yes, this actually bothers me as well. Originally I set it up this way to follow the pattern set up by the product which had Train and Edit as distinct code files, but I agree that we can refactor this into something that makes more sense from the test code perspective. Adding it to my todo list and I'll keep it in mind on how best to handle this change. So for this PR we will leave it alone.